### PR TITLE
Jhbertra/addresses

### PR DESCRIPTION
**Pull request by jhbertra**
Updates the create and redeem endpoints to accept Shelley Addresses. This is a bit of a hack, and support for addresses should be properly pushed upstream into Plutus.

### CODE REVIEW

File: marlowe/src/Language/Marlowe/Client.hs
Comment (bwbush) [inline marlowe/src/Language/Marlowe/Client.hs:136]: `AddressInEra ShelleyEra` should work. We may want to adjust this eventually, when for instance we allow scripts to hold role tokens.

File: marlowe/src/Language/Marlowe/Client.hs
Comment (bwbush) [inline marlowe/src/Language/Marlowe/Client.hs:485]: We might consider a PR to move these into `plutus-apps`, but the change would involve adding a constructor to the constraint type there.

File: marlowe/src/Language/Marlowe/Client.hs
Comment (bwbush) [inline marlowe/src/Language/Marlowe/Client.hs:518]: It actually wouldn't be difficult to support Byron addresses, the `Cardano.Api.Byron` module has address functions similar to `Cardano.Api.Shelley`.

**Review by bwbush — APPROVED** [review]
Looks good!

I'll need to adjust some of the still non-automated tests for the PAB, but that will be trivial, and I can do that after this is merged.

File: marlowe/src/Language/Marlowe/Client.hs
Comment (jhbertra) [reply marlowe/src/Language/Marlowe/Client.hs:136]: `AddressInEra` has a `FromJSON` instance, `AddressAny` doesn't

File: marlowe/src/Language/Marlowe/Client.hs
Comment (jhbertra) [reply marlowe/src/Language/Marlowe/Client.hs:518]: I'll merge this PR and we can add it if/when we need it

**Pull request merged**
